main.go

```diff
@@ -180,46 +193,35 @@ func setupControllers(mgr ctrl.Manager, cCache *cache.Cache, queues *queue.Manag
 		os.Exit(1)
 	}
 
-	if isFrameworkEnabled(cfg, job.FrameworkName) {
-		if err := job.NewReconciler(mgr.GetScheme(),
-			mgr.GetClient(),
-			mgr.GetEventRecorderFor(constants.JobControllerName),
-			jobframework.WithManageJobsWithoutQueueName(manageJobsWithoutQueueName),
-			jobframework.WithWaitForPodsReady(waitForPodsReady(cfg)),
-		).SetupWithManager(mgr); err != nil {
-			setupLog.Error(err, "unable to create controller", "controller", "Job")
-			os.Exit(1)
-		}
-		if err := job.SetupWebhook(mgr, jobframework.WithManageJobsWithoutQueueName(manageJobsWithoutQueueName)); err != nil {
-			setupLog.Error(err, "Unable to create webhook", "webhook", "Job")
-			os.Exit(1)
-		}
-	} else {
-		if err := noop.SetupWebhook(mgr, job.WebhookType()); err != nil {
-			setupLog.Error(err, "Unable to create webhook", "webhook", "Job")
-			os.Exit(1)
-		}
+	opts := []jobframework.Option{
+		jobframework.WithManageJobsWithoutQueueName(manageJobsWithoutQueueName),
+		jobframework.WithWaitForPodsReady(waitForPodsReady(cfg)),
 	}
-
-	if isFrameworkEnabled(cfg, mpijob.FrameworkName) {
-		if err := mpijob.NewReconciler(mgr.GetScheme(),
-			mgr.GetClient(),
-			mgr.GetEventRecorderFor(constants.KueueName+"-mpijob-controller"),
-			jobframework.WithManageJobsWithoutQueueName(manageJobsWithoutQueueName),
-			jobframework.WithWaitForPodsReady(waitForPodsReady(cfg)),
-		).SetupWithManager(mgr); err != nil {
-			setupLog.Error(err, "unable to create controller", "controller", "MPIJob")
-			os.Exit(1)
-		}
-		if err := mpijob.SetupMPIJobWebhook(mgr, jobframework.WithManageJobsWithoutQueueName(manageJobsWithoutQueueName)); err != nil {
-			setupLog.Error(err, "Unable to create webhook", "webhook", "MPIJob")
-			os.Exit(1)
-		}
-	} else {
-		if err := noop.SetupWebhook(mgr, mpijob.WebhookType()); err != nil {
-			setupLog.Error(err, "Unable to create webhook", "webhook", "MPIJob")
-			os.Exit(1)
+	err := jobframework.ForEachIntegration(func(name string, cb jobframework.IntegrationCallbacks) error {
+		log := setupLog.WithValues("integration", name)
+		if isFrameworkEnabled(cfg, name) {
+			if err := cb.NewReconciler(mgr.GetScheme(),
+				mgr.GetClient(),
+				mgr.GetEventRecorderFor(name+"-"+constants.KueueName+"-controller"),
+				opts...,
+			).SetupWithManager(mgr); err != nil {
+				log.Error(err, "unable to create controller")
+				return err
+			}
+			if err := cb.SetupWebhook(mgr, opts...); err != nil {
+				setupLog.Error(err, "Unable to create webhook", "webhook", "Job")
```

Comment:
```suggestion
				setupLog.Error(err, "Unable to create webhook", "webhook", frameworkName)
```

Reply:
Oh, I was missing :)

If so, we don't need "Job". This means the below:

```suggestion
				setupLog.Error(err, "Unable to create webhook", "webhook")
```